common/Util.hpp

```diff
@@ -886,6 +886,24 @@ int main(int argc, char**argv)
         return std::string(message, size);
     }
 
+    inline
+    std::string eliminatePrefix(const std::string& str, const std::string& prefix)
+    {
+        std::string::const_iterator prefix_pos;
+        std::string::const_iterator str_pos;
+
+        std::tie(prefix_pos,str_pos)= std::mismatch(prefix.begin(),prefix.end(),str.begin());
+
+        if( prefix_pos == prefix.end() )
+        {
+            return std::string(str_pos,str.end());
+        }
+        else
+        {
+            return std::string();
```

Comment:
This function needs a comment to describe what its semantics are. Without reading the code, I would not expect it to return an empty string if str doesn't start with prefix.

Reply:
Also, compared to the startsWith() functions in the same file, this is much different in style. (Those use memcmp() directly on the string bytes.) That is probably not a problem, just a slight inconsistency.

Reply:
A one line description of eliminate prefix added